include/cutlass/gemm/device/gemm.h

```diff
@@ -362,7 +362,22 @@ class Gemm {
     if (!kSplitKSerial && args.split_k_slices > 1) {
       return Status::kErrorInvalidProblem;
     }
+    
+    // MaxDynamicSharedMemorySize must be smaller than the device attribute cudaDevAttrMaxSharedMemoryPerBlockOptin minus the function attribute sharedSizeBytes  
+    int smem_size = static_cast<int>(sizeof(typename GemmKernel::SharedStorage));
+    cudaDeviceProp properties;
+    result = cudaGetDeviceProperties(&properties, device_idx);
+
+    if (result != cudaSuccess) {
+      return Status::kErrorMemoryAllocation;
+    }
+
+    int smem_capacity = static_cast<int>(properties.sharedMemPerMultiprocessor);
```

Comment:
we need to use `sharedMemPerBlockOptin` to get the max usable shared memory.

Reply:
@hwu36 Thanks for reviewing this. I can't find the `sharedMemPerBlockOptin` attribute in the device properties, they only offer the more general `sharedMemPerBlock`. 
Would it make sense to go back to my original suggestion of getting the attributes through `cudaDeviceGetAttribute` and `cudaFuncGetAttributes`?  At least this is what I understood from the documentation of the `cudaFuncSetAttribute` function.

Reply:
@alexfreudenberg https://docs.nvidia.com/cuda/cuda-runtime-api/structcudaDeviceProp.html#structcudaDeviceProp_1c23eacaeeb59e1798d6d442be70f1c3e